npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -172,12 +172,13 @@ def run_npm_subdependency_updater(lockfile_name:)
         # rubocop:disable Metrics/PerceivedComplexity
         # rubocop:disable Metrics/MethodLength
         def handle_npm_updater_error(error, lockfile)
-          if error.message.match?(MISSING_PACKAGE)
-            package_name =
-              error.message.match(MISSING_PACKAGE).
-              named_captures["package_req"].
-              gsub("%2f", "/")
-            handle_missing_package(package_name, error, lockfile)
+          error_message = error.message
+          if error_message.match?(MISSING_PACKAGE)
+            sanitized_error, package_name = sanitized_error_package_name(
+              error_message,
+              error_message.match(MISSING_PACKAGE).named_captures["package_req"]
```

Comment:
Can you do this calculation outside of the function arguments, and give it a name, so it's clearer what's going into the function?

Reply:
I also find it really surprising that this function returns an array. Can you refactor it so it's two separate functions or something like that?

Reply:
Yes good push!